README.md

```diff
@@ -1,2 +1,302 @@
+
+### Create the `ExportSettings`:
+
+Use `QMLSTYLE` for the export of the qml stylefile.
+Use `DEFINITION` to export the qlr definition file.
+USE `SOURCE` to store the source in the YAML tree.
+
+The QgsLayerTreeNode or the layername can be used as key.
+
+```py
+export_settings = ExportSettings()
+export_settings.set_setting_values(
+    type = ExportSettings.ToppingType.QMLSTYLE, node = None, name = "Street", export = True
+)
+export_settings.set_setting_values(
+    type = ExportSettings.ToppingType.SOURCE, node = None, name = "Park", export = True
+)
+export_settings.set_setting_values(
+    type = ExportSettings.ToppingType.DEFINITION, node = None, name = "Info Layers", export = True
+)
+export_settings.set_setting_values(
+    type = ExportSettings.ToppingType.SOURCE, node = None, name = "Landeskarten (grau)", export = True
+)
+```
+
+Additionally you can pass category flags `QgsMapLayer.StyleCategories` to define what categories needs to be included in the QML Stylefile:
+
+```py
+category_flags = QgsMapLayer.StyleCategory.AllStyleCategories
+
+export_settings.set_setting_values(
+    type = ExportSettings.ToppingType.QMLSTYLE, node = None, name = "Street", export = True, categories = category_flags
+)
+```
+
+### Generate the Files for a `ProjectTopping` containing `ExportSetting`
+When parsing the QgsProject we need to pass the `ExportSettings`:
+```
+
+project_topping.parse_project(project, export_settings)
+project_topping.generate_files(target)
+```
+
+Structure looks like this:
+
+```
+repo
+└── freddys_qgis_topping
+    ├── layerdefinition
+    │   └── freddys_qgis_project_info_layers.qlr
+    └── projecttopping
+        └── freddys_qgis_project.yaml
+    └── layerstyle
+        └── freddys_qgis_project_street.qml
+```
+
+And the YAML looks like this:
+
+```yaml
+layerorder: []
+layertree:
+- Street:
+    checked: true
+    expanded: true
+    stylefile: freddys_qgis_topping/layerstyle/freddys_qgis_project_street.qml
+- Park:
+    checked: false
+    expanded: true
+    provider: ogr
+    uri: /home/freddy/qgis_projects/bakery/cityandcity.gpkg|layername=park
+- Building:
+    checked: true
+    expanded: true
+- Info Layers:
+    checked: true
+    definitionfile: freddys_qgis_topping/layerdefinition/freddys_qgis_project_info_layers.qlr
+    expanded: true
+    group: true
+- Background:
+    checked: true
+    child-nodes:
+    expanded: true
+    group: true
+    - Landeskarten (grau):
+        checked: true
+        expanded: true
+        provider: wms
+        uri: contextualWMSLegend=0&crs=EPSG:2056&dpiMode=7&featureCount=10&format=image/jpeg&layers=ch.swisstopo.pixelkarte-grau&styles&url=https://wms.geo.admin.ch/?%0ASERVICE%3DWMS%0A%26VERSION%3D1.3.0%0A%26REQUEST%3DGetCapabilities
+```
+
+## Infos for Devs
+
+### Code style
+
+Is enforced with pre-commit. To use, make:
+```
+pip install pre-commit
```

Comment:
Good idea but still todo I think?

Reply:
Do you mean because the code is not in black format? Because the pre-commit should be set up. But the code has been commited first. Will make a reformat in the end.